crates/autopilot/src/database/onchain_order_events.rs

```diff
@@ -425,61 +429,74 @@ type GeneralOnchainOrderPlacementData = (
     OnchainOrderPlacement,
     Order,
 );
-async fn parse_general_onchain_order_placement_data(
-    quoter: &dyn OrderQuoting,
-    contract_events_and_quotes_zipped: Vec<(EthContractEvent<ContractEvent>, i64, i64)>,
+async fn parse_general_onchain_order_placement_data<'a>(
+    quoter: &'a dyn OrderQuoting,
+    order_placement_events_and_quotes_zipped: Vec<(EthContractEvent<ContractEvent>, i64, i64)>,
     domain_separator: DomainSeparator,
     settlement_contract: H160,
-    liquidity_order_owners: &HashSet<H160>,
+    liquidity_order_owners: &'a HashSet<H160>,
+    metrics: &'static Metrics,
 ) -> Vec<GeneralOnchainOrderPlacementData> {
-    let futures = contract_events_and_quotes_zipped.into_iter().map(
-        |(EthContractEvent { data, meta }, event_timestamp, quote_id)| async move {
-            let meta = match meta {
-                Some(meta) => meta,
-                None => return Err(anyhow!("event without metadata")),
-            };
-            let event = match data {
-                ContractEvent::OrderPlacement(event) => event,
-                _ => {
-                    return Err(anyhow!(
-                        "parse_general_onchain_order_placement_data should not reach this state"
-                    ))
+    let futures = order_placement_events_and_quotes_zipped.into_iter().map(
+        |(EthContractEvent { data, meta }, event_timestamp, quote_id)| {
+            let metrics_clone = metrics.clone();
```

Comment:
Last nit to clear this diff up:
`metrics` doesn't have to be cloned into the future which means the whole block does not have to be reformatted due to the increased level of indentation. This should make the diff way smaller.